Quality: PowerShell Summit GH-3400 (#3401)
MicrosoftDocs/windows-powershell-docs: docset/winserver2022-ps/addsdeployment/Test-ADDSDomainControllerUninstallation.md

# Test-ADDSDomainControllerUninstallation

## SYNOPSIS

> 

Reviewer: ```suggestion
```

Runs the prerequisites for uninstalling a domain controller in Active Directory.

## SYNTAX

### ADDSDomainControllerUninstall (Default)

```
Test-ADDSDomainControllerUninstallation [-LocalAdministratorPassword <SecureString>]
 [-Credential <PSCredential>] [-DemoteOperationMasterRole]
 [-DnsDelegationRemovalCredential <PSCredential>] [-IgnoreLastDCInDomainMismatch]
 [-IgnoreLastDnsServerForZone] [-LastDomainControllerInDomain] [-NoRebootOnCompletion]
 [-RemoveApplicationPartitions] [-RemoveDnsDelegation] [-RetainDCMetadata] [-Force]
 [<CommonParameters>]
```

### ADDSDomainControllerUninstallForceRemoval

```
Test-ADDSDomainControllerUninstallation [-LocalAdministratorPassword <SecureString>]
 [-Credential <PSCredential>] [-DemoteOperationMasterRole] [-ForceRemoval]
 [-NoRebootOnCompletion] [-Force] [<CommonParameters>]
```

## DESCRIPTION

> The `Test-ADDSDomainControllerUninstallation` cmdlet runs those prerequisite checks which would be
> performed if you were to use the Uninstall-ADDSDomainController cmdlet to uninstall a domain
> controller in Active Directory. It differs from using the `-WhatIf` parameter with the
> `Uninstall-ADDSDomainController` cmdlet in that instead of summarizing the changes that would occur
> during the uninstallation process, this cmdlet actually tests whether those changes are possible
> given the current environment.
> 
> For more information on the scope of these prerequisite checks that the ADDSDeployment module
> performs when using this cmdlet see the section ADPrep and Prerequisite Checking Architecture in
> [AD DS Simplified Administration](https://go.microsoft.com/fwlink/?LinkID=237244).

Reviewer: ```suggestion
The `Test-ADDSDomainControllerUninstallation` cmdlet runs those prerequisite checks which would be
performed if you were to use the `Uninstall-ADDSDomainController` cmdlet to uninstall a domain
controller in Active Directory. It differs from using the **WhatIf** parameter with the
`Uninstall-ADDSDomainController` cmdlet in that instead of summarizing the changes that would occur
during the uninstallation process, this cmdlet actually tests whether those changes are possible
given the current environment.

For more information on the scope of these prerequisite checks that the **ADDSDeployment** module
performs when using this cmdlet see the section ADPrep and Prerequisite Checking Architecture in
[AD DS Simplified Administration](https://go.microsoft.com/fwlink/?LinkID=237244).
```

## EXAMPLES

### Example 1: Test if uninstalling a domain controller is possible

```powershell
Test-ADDSDomainControllerUninstallation
```

This command runs the prerequisites to determine if the uninstall of an additional domain controller
in a domain is possible. The command also prompts the user to set and confirm the local
Administrator password prior to completing the uninstallation process.

## PARAMETERS

### -Credential

> Specifies the user name and password that corresponds to the account used to install the domain
> controller. To prompt the user to supply a password, use Runs the prerequisites (only) to determine
> if installing a domain controller is possible that includes a DNS server for the corp.contoso.com
> domain, using domain administrator credentials, and then prompts the user to correctly specify the
> Directory Services Restore Mode (DSRM) password. Use the `Get-Credential` cmdlet in place of an
> existing _PSCredential_ type. This parameter will cause Windows PowerShell to prompt the user to
> enter credentials using the Windows security login UI.

Reviewer: ```suggestion
Specifies the user name and password that corresponds to the account used to install the domain
controller. To prompt the user to supply a password, use Runs the prerequisites (only) to determine
if installing a domain controller is possible that includes a DNS server for the `corp.contoso.com`
domain, using domain administrator credentials, and then prompts the user to correctly specify the
Directory Services Restore Mode (DSRM) password. Use the `Get-Credential` cmdlet in place of an
existing **PSCredential** type. This parameter will cause Windows PowerShell to prompt the user to
enter credentials using the Windows security login UI.
```

```

### -DemoteOperationMasterRole

Indicates that forced demotion should continue even if an operations master role is discovered on

> domain controller from which AD DS is being removed.

Reviewer 2: ```suggestion
the domain controller from which AD DS is being removed.
```

```

### -DnsDelegationRemovalCredential

> Specifies the account credentials to use when you create or remove the DNS delegation. If you do not
> specify a value, the account credentials that you specify for the AD DS installation or removal are
> used to remove the DNS delegation. As an alternative, you can specify the asterisk (*) to prompt the
> user to enter credentials.

Reviewer: ```suggestion
Specifies the account credentials to use when you create or remove the DNS delegation. If you do
not specify a value, the account credentials that you specify for the AD DS installation or removal
are used to remove the DNS delegation. As an alternative, you can specify the asterisk (`*`) to
prompt the user to enter credentials.
```

```

### -IgnoreLastDCInDomainMismatch

> Indicates that Windows PowerShell ignores any inconsistency that it detects with the value that you
> specify for the `-LastDomainControllerInDomain` parameter. For instance, if you specify
> `-LastDomainControllerInDomain` but Windows PowerShell detects that there is actually another active
> domain controller in the domain, you can specify the `-IgnoreLastDCInDomainMismatch` parameter to
> have Windows PowerShell continue the removal of AD DS from the domain controller despite the
> inconsistency that it has detected. Similarly, if you do not specify `-LastDomainControllerInDomain`
> but Windows PowerShell is unable to detect that another domain controller is in the domain, you can
> specify `-IgnoreLastDCInDomainMismatch` to have Windows PowerShell continue to remove AD DS from the
> domain controller.

Reviewer: ```suggestion
Indicates that Windows PowerShell ignores any inconsistency that it detects with the value that you
specify for the **LastDomainControllerInDomain** parameter. For instance, if you specify
**LastDomainControllerInDomain** but Windows PowerShell detects that there is actually another
active domain controller in the domain, you can specify the **IgnoreLastDCInDomainMismatch**
parameter to have Windows PowerShell continue the removal of AD DS from the domain controller
despite the inconsistency that it has detected. Similarly, if you do not specify
**LastDomainControllerInDomain** but Windows PowerShell is unable to detect that another domain
controller is in the domain, you can specify **IgnoreLastDCInDomainMismatch** to have Windows
PowerShell continue to remove AD DS from the domain controller.
```

```

### -LocalAdministratorPassword

Specifies a local administrator account password when AD DS is removed from a domain controller. In
earlier releases, where uninstall of AD DS was done using `Dcpromo.exe` for demotion, the default

> was to allow an empty password for this setting. In Windows PowerShell, the ADDS Deployment module

Reviewer 2: ```suggestion
was to allow an empty password for this setting. In Windows PowerShell, the AD DS Deployment module
```

```

### -RemoveDnsDelegation

Indicates that the cmdlet preserves DNS delegations that point to this DNS server from the parent
DNS zone.

> By default, this parameter is set to false, which means DNS delegations that point to this server
> from the parent DNS zone will not be retained after uninstallation of the domain controller. This
> setting corresponds to the earlier `Dcpromo.exe` parameter default of `/RemoveDNSDelegation:Yes`.

Reviewer: ```suggestion
By default, this parameter is set to `$false`, which means DNS delegations that point to this
server from the parent DNS zone will not be retained after uninstallation of the domain controller.
This setting corresponds to the earlier `Dcpromo.exe` parameter default of
`/RemoveDNSDelegation:Yes`.
```
